### ISSUE TITLE
[Write | EmailConfirmation] create slice RequestEmailConfirmation -> EmailConfirmationWasRequested

### ISSUE THREAD

**Issue by nowakprojects** [CodersCrew/coderscamp#279]
**Context**
This implementation provides a generic approach to confirming any operation by email.
Current use case is user registration confirmation. But the same approach may be used in similar use cases like you approve bank account operations with sms code. 
Physical sending of email message by SMTP is not part of this task.

**Dependencies (blockers)**
- [ ] Make sure that #278 is done (must be merged before)

**Subtasks**
If you need more info or have questions, fell free to ask @nowakprojects.

Command definition:
- [ ] Create command type `RequestEmailConfirmation` in `modules/shared/commands`
- [ ] Create `RequestEmailConfirmationApplicationCommand` class in `modules/shared/commands`

Domain logic:
- [ ] Create function requestEmailConfirmation in `modules/write/email-confirmation/domain`.
- [ ] Return EmailConfirmationWasRequested from the method (no domain rules so far, will be introduced in next tasks).

Application logic:
- [ ] Create CommandHandler for RequestEmailConfirmationApplicationCommand
- [ ] Execute domain logic. Event stream name should be in format `EmailConfirmation_${userId}_${confirmationFor}` _Do you know why? Try to elaborate before taking this task._

Presentation logic:
- Rest API endpoint will be done in another task.

NestJS Module:
- [ ] Create EmailConfirmationWriteModule module and declare all required providers. Similar as other modules.

**Event Modeling**
MIRO link: https://miro.com/app/board/o9J_lQvnN28=/?moveToWidget=3074457363394933027&cot=14
